integrations/gradle-bloop/src/main/scala/bloop/integrations/gradle/tasks/BloopInstallTask.scala

```diff
@@ -0,0 +1,63 @@
+package bloop.integrations.gradle.tasks
+
+import java.io.File
+import java.nio.file.Files
+
+import bloop.integrations.gradle.BloopParameters
+import bloop.integrations.gradle.model.BloopConverter
+import bloop.integrations.gradle.syntax._
+import org.gradle.api.tasks.{SourceSet, TaskAction}
+import org.gradle.api.{DefaultTask, Project}
+
+/**
+ * Define a Gradle task that generates bloop configuration files from a Gradle project.
+ *
+ * This part of the plugin is mainly in charge of handling source sets. Source sets are logical
+ * group of sources and resources ([[https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html]]).
+ *
+ * Source sets are the equivalent of sbt configurations (approximately). The default java plugin
+ * adds two source sets by default (compile and test) [[https://docs.gradle.org/current/userguide/java_plugin.html]].
+ */
+final class BloopInstallTask extends DefaultTask with TaskLogging {
+  private val project: Project = getProject
+  private val parameters: BloopParameters = project.getExtension[BloopParameters]
+  private val converter = new BloopConverter(parameters)
+
+  @TaskAction
+  def run(): Unit = {
+    val targetDir: File = parameters.targetDir
+    info(s"Generating Bloop configuration to ${targetDir.getAbsolutePath}")
+
+    if (!targetDir.exists()) {
+      debug(s"Creating target directory ${targetDir}")
+      Files.createDirectory(targetDir.toPath)
+    }
+
+    val mainSourceSet = project.getSourceSet(parameters.mainSourceSet)
+    val otherSourceSets = project.allSourceSets.filter(_.getName != parameters.mainSourceSet)
+
+    // The 'main' source set maps to the raw project name (as all integrations do)
+    val mainProjectName = converter.getProjectName(project, mainSourceSet)
+    generateBloopConfiguration(mainProjectName, Set.empty, mainSourceSet, targetDir)
+
+    // Generate the bloop configuration files for the rest of the source sets
+    for (sourceSet <- otherSourceSets) {
+      val projectName = converter.getProjectName(project, sourceSet)
+      // Hardcore an implicit dependency for every source set to the main source set (compile)
+      generateBloopConfiguration(projectName, Set(mainProjectName), sourceSet, targetDir)
```

Comment:
I'm surprised that this is not present in the project configuration of gradle. Would you mind confirming that adding this dependency is necessary?

Reply:
What I found so far is:
- The `test` source set's compile classpath is extended with the `main` source set's compilation output: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java#L277
- The `test` configuration "extends from" the `main` configuration: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java#L427

So it's not as high level as I would expect but hopefully can be somehow used to figure out the depenencies. Will experiment with it.

Reply:
After thinking more about this, we do need to add this implicit dependency as a project dependency. As you have figured out, Gradle adds the main output dir to the test classpath, but at the project level we still need to add this dependency so that Bloop can have properly schedule compilation sequentially (if there was no dependency, Bloop would run the compilation of main and test separately, which would fail because test depends on main).

Reply:
Yes that was my orgiginal intention too. But it would be nice figuring out the dependencies between the Gradle source sets instead of hardcoding that all non-main depends on main.

Reply:
Yeah. That’s most likely not supported by Gradle, though... Maybe exposing a way to configure that in the gradle config would make sense.